boot/boot.js

```diff
@@ -510,26 +537,46 @@ $tw.utils.evalGlobal = function(code,context,filename) {
 		contextValues.push(value);
 	});
 	// Add the code prologue and epilogue
-	code = "(function(" + contextNames.join(",") + ") {(function(){\n" + code + "\n;})();\nreturn exports;\n})\n";
+	code = [
+		(!$tw.browser ? polyfill : ""),
+		"(function(" + contextNames.join(",") + ") {",
+		"  (function(){\n" + code + "\n;})();",
+		(!$tw.browser && sandbox && !allowGlobals) ? globalCheck : "",
+		"  return exports;\n",
+		"})"
+	].join("\n");
+	// code = "(function(" + contextNames.join(",") + ") {(function(){  \n" + code + "\n;})();\nreturn exports;\n})\n";
 	// Compile the code into a function
 	var fn;
 	if($tw.browser) {
 		fn = window["eval"](code + "\n\n//# sourceURL=" + filename);
 	} else {
-		fn = vm.runInThisContext(code,filename);
+		if(sandbox){
+			fn = vm.runInContext(code,sandbox,filename)
+		} else {
+			fn = vm.runInThisContext(code,filename);
+		}
 	}
 	// Call the function and return the exports
 	return fn.apply(null,contextValues);
 };
-
+$tw.utils.sandbox = !$tw.browser ? vm.createContext({}) : undefined; 
```

Comment:
It might be safer to check for `$tw.node`; TW 5 also runs under nw.js where both `$tw.browser` and `$tw.node` are `true` (and Node apis are available).

Reply:
That's based on evalSandboxed, which uses `$tw.browser`, so this needs to be the opposite of that.